### PR TITLE
Add ability to change vehicle tire using hydraulic muscles bionic instead jack.

### PR DESCRIPTION
**Pull request by DZiral**
Hydraulic muscles give big strength boost to character. So I think it is possible to use all this strength to replace car tire without jack.

### CODE REVIEW

**Comment by Barhandar** [comment]
Why not make it generic "X or more strength allows to lift a car to change the tire"? X being 24 in this case, since 4 is minimum creation-stat and hydraulic muscles add 20.

**Comment by DZiral** [comment]
Yes, look like check strength more universal idea. I think I try to do it.

**Comment by Coolthulhu** [comment]
Bonus points for it actually depending on weight, so that wooden framed bike can be lifted without hydraulic muscles and a deathroller can't be even with.

**Comment by DZiral** [comment]
May be use something like this: vehicle must be lighter then Strength * 40kg (dont need lift car completely, only lift one side enough for tire change)
Average strength 14 grant us 560 kg car weight allowance. With this strength it is possible to change tires in motorcycles and not possible to standard cars. With bionics it is possible to work with normal cars and not trucks/armored cars/rollers.

**Comment by DZiral** [comment]
Something like this

File: src/veh_interact.cpp [hunk 1]
```diff
@@ -1165,6 +1167,7 @@ void veh_interact::do_tirechange()
     display_mode('c');
     werase( w_msg );
     int msg_width = getmaxx(w_msg);
+    bool has_str = (g->u.get_str() * TIRE_CHANGE_STR_MOD) >= veh->total_mass();
```
Comment (Coolthulhu) [inline src/veh_interact.cpp]: How about this: splitting the value into
```
const int needed_strength = veh->total_mass() / TIRE_CHANGE_STR_MOD;
const bool has_str = g->u.get_str() >= needed_strength;
```
And then below, in the print message you could say how much strength is required to replace the jack.

File: src/veh_interact.cpp [hunk 2]
```diff
@@ -447,7 +448,8 @@ task_reason veh_interact::cant_do (char mode)
         break;
     case 'c': // change tire
         valid_target = wheel != NULL;
-        has_tools = has_wrench && has_jack && has_wheel;
+		has_str = (g->u.get_str() * TIRE_CHANGE_STR_MOD) >= veh->total_mass();
```
Comment (Coolthulhu) [inline src/veh_interact.cpp]: A tab sneaked in here. Please use 4 spaces instead of tabs.

**Comment by DZiral** [comment]
Change code to display needed strength and fix tabs.

File: src/veh_interact.cpp [hunk 3]
```diff
@@ -1172,9 +1176,10 @@ void veh_interact::do_tirechange()
         return;
     case LACK_TOOLS:
         fold_and_print(w_msg, 0, 1, msg_width - 2, c_ltgray,
-                       _("To change a wheel you need a <color_%1$s>wrench</color> and a <color_%2$s>jack</color>."),
+                       _("To change a wheel you need a <color_%1$s>wrench</color> and a <color_%2$s>jack</color>. Having strength %3$d or more allows to lift this car to change the tire."),
```
Comment (Coolthulhu) [inline src/veh_interact.cpp]: The description is a bit misleading. How about something like "you need a wrench and either a jack or x str"?

**Comment by DZiral** [comment]
Changed description of tire lack tools change message, also add color to strength value.

**Comment by Coolthulhu** [comment]
Later on it would be cool if it wasn't just for tire changing, but also for installing and removing wheels.

**Comment by DZiral** [comment]
Add same changes to removing wheels in this pull https://github.com/CleverRaven/Cataclysm-DDA/pull/13185

**Pull request merged**